### DISCUSSION TITLE
Extension problem: Copilot and solidity doesn't work correctly

### DISCUSSION THREAD

**Discussion by aliqbdoli** [Cyfrin/foundry-full-course-cu#2465] [General]
I'm using vscode and the copilot extension doesn't work in my country, also when I install the Solidity extension it doesn't recognize many variables and I have to type manually and this causes syntax errors, is there a solution?  Is there an alternative extension?

**Comment by EngrPips** [comment]
What solidity extension did you install?

**Reply by aliqbdoli** [reply]
Copilot 
Solidity hard hat
Tanine

**Reply by EngrPips** [reply]
I don't use any AI assistance when coding, so I am not sure which one is good, but you should try the below for the solidity file editor. 

<img width="1440" alt="Screenshot 2024-09-13 at 16 26 05" src="https://github.com/user-attachments/assets/bd70a047-3871-41ce-992c-ca333bbd5e94">

**Comment by aliqbdoli** [comment]
I installed that extension but usually it doesn't work correctly, for
example After defining a constant variable, when using it in the function,
by default, I write its name first, that variable is not recognized, and I
have to copy the name of the variable and then use it in the functions, but
some words are recognized, such as the word  Different functions or types

**Reply by EngrPips** [reply]
I don't know why it is not working correctly.  It works fine for me.

**Comment by aliqbdoli** [comment]
Thank  you for helping, I had several extensions installed for solidity and
copilot and they were not compatible, finally I deleted them all and
installed only the solidity extension you showed and it solved the problem

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.

**Comment by aliqbdoli** [comment]
Thanks, I will.
